src/Paramore.Brighter.Tranformers.AWS/ServiceCollectionExtensions.cs

```diff
@@ -0,0 +1,66 @@
+﻿#region Licence
+/* The MIT License (MIT)
+Copyright © 2022 Ian Cooper <ian_hammond_cooper@yahoo.co.uk>
+
+Permission is hereby granted, free of charge, to any person obtaining a copy
+of this software and associated documentation files (the “Software”), to deal
+in the Software without restriction, including without limitation the rights
+to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+copies of the Software, and to permit persons to whom the Software is
+furnished to do so, subject to the following conditions:
+
+The above copyright notice and this permission notice shall be included in
+all copies or substantial portions of the Software.
+
+THE SOFTWARE IS PROVIDED “AS IS”, WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+THE SOFTWARE. */
+ 
+#endregion
+
+using System;
+using System.Net.Http;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.DependencyInjection.Extensions;
+using Paramore.Brighter.Transforms.Storage;
+
+namespace Paramore.Brighter.Tranformers.AWS
+{
+    public static class ServiceCollectionExtensions
+    {
+        /// <summary>
+        /// Will add an S3 Luggage Store into the Service Collection
+        /// </summary>
+        /// <param name="services"></param>
+        /// <param name="configure"></param>
+        public static void AddS3LuggageStore(this IServiceCollection services, Action<S3LuggageOptions> configure)
```

Comment:
Should we be adding and retrieving these things from the ServiceCollection?

Given that we usually ensure that you can BYO DI, I feel this should be in the Brighter Builder

Reply:
My Mistake, I see this can work with BYO DI (you just have to implement the Transform Factory for this), Something still feels wrong to me around it being off of Service Collection, but maybe it just needs to sit with me a little long :)

Reply:
Well this is really an optimization for folks using ServiceCollection, which we have elsewhere. In essence folks on other IoC containers need to implement an equivalent